facade/src/main/java/org/jboss/pnc/facade/providers/SCMRepositoryProviderImpl.java

```diff
@@ -71,9 +117,120 @@ public SCMRepositoryProviderImpl(RepositoryConfigurationRepository repository,
      * @param <T>
      */
     private <T> void addToListIfStringNotNullAndNotEmpty(List<T> list, String str, T item) {
-
         if (str != null && !str.isEmpty()) {
             list.add(item);
         }
     }
+
+    @Override
+    public RepositoryCreationResponse createSCMRepository(String scmUrl, Boolean preBuildSyncEnabled, Consumer<Integer>... consumers) {
+        log.trace("Received request to start RC creation with url autodetect: " + scmUrl + " (sync enabled? " + preBuildSyncEnabled + ")");
+        if(StringUtils.isEmpty(scmUrl)) throw new InvalidEntityException("You must specifyt the SCM URL.");
+
+        if(scmUrl.contains(config.getInternalScmAuthority())){
+            SCMRepository scmRepository = getInternalRepository(scmUrl);
+            for (Consumer<Integer> consumer : consumers) {
+                consumer.accept(scmRepository.getId());
+            }
+            return new RepositoryCreationResponse(scmRepository);
+        }else{
+            RepositoryCreationTask task = getExternalRepository(scmUrl, preBuildSyncEnabled);
+            for (Consumer<Integer> consumer : consumers) {
+                task.addListener(
```

Comment:
I don't know if adding listeners after starting listeners is good practice. Could theoretically create race conditions.